daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Util.scala

```diff
@@ -100,4 +104,112 @@ object Util {
       case None => Right(None)
     }
 
+  /** Compute the inputs to a DAML transaction (in traversal order), that is, the referenced contracts, parties,
+    *  and key.
+    */
+  def collectInputs(
+      tx: Transaction.Transaction,
+      addContract: ContractId => Unit,
+      addParty: Ref.Party => Unit,
+      addGlobalKey: GlobalKey => Unit,
+  ): Unit = {
```

Comment:
I'm not a big fan of `Util` classes/objects because they tend to become a dumping ground for any random function.

I would prefer if this function lived in its own object, as before.

Reply:
done.

Reply:
Thanks!